src/app/pages/FastBtcPage/components/Deposit/AddressForm.tsx

```diff
@@ -1,16 +1,45 @@
-import React, { useContext } from 'react';
+import React, { useCallback, useContext } from 'react';
 import { DepositDetails } from './DepositDetails';
 import { AddressQrCode, URIType } from 'app/components/Form/AddressQrCode';
 import { DepositContext } from '../../contexts/deposit-context';
+import { FastBtcDirectionType } from '../../types';
+import { FastBtcButton } from '../FastBtcButton';
+import { useTranslation } from 'react-i18next';
+import { translations } from 'locales/i18n';
+import { useTransak } from 'app/components/TransakDialog/useTransak';
 
-export const AddressForm: React.FC = () => {
+type AddressFormProps = {
+  type: FastBtcDirectionType;
+};
+
+export const AddressForm: React.FC<AddressFormProps> = ({ type }) => {
   const { address } = useContext(DepositContext);
+  const { t } = useTranslation();
+  const { handleClick, isWrongChainId } = useTransak();
+
+  const handleTransakClick = useCallback(() => handleClick('BTC', address), [
+    address,
+    handleClick,
+  ]);
 
   return (
     <>
-      <div className="tw-full tw-mb-8">
+      <div className="tw-w-full tw-mb-8">
         <DepositDetails />
-        <AddressQrCode uri={URIType.BITCOIN} address={address} />
+        <AddressQrCode
+          uri={URIType.BITCOIN}
+          address={address}
+          hideQr={type === FastBtcDirectionType.TRANSAK}
+        />
+
+        {type === FastBtcDirectionType.TRANSAK && (
```

Comment:
Extract this condition to a variable (e.g. `isTransakDeposit`), use `useMemo` and use it here and on the line `32` as well.